core/src/main/kotlin/behavior/channel/threads/ThreadChannelBehavior.kt

```diff
@@ -0,0 +1,129 @@
+package dev.kord.core.behavior.channel.threads
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.common.exception.RequestException
+import dev.kord.core.Kord
+import dev.kord.core.behavior.channel.MessageChannelBehavior
+import dev.kord.core.cache.data.toData
+import dev.kord.core.entity.channel.Channel
+import dev.kord.core.entity.channel.thread.ThreadChannel
+import dev.kord.core.entity.channel.thread.ThreadUser
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.rest.builder.channel.thread.ThreadModifyBuilder
+import kotlinx.coroutines.flow.Flow
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+interface ThreadChannelBehavior : MessageChannelBehavior {
+    /**
+     * Requests to get all members of the current thread.
+     *
+     * The returned flow is lazily executed, any [RequestException] will be thrown on
+     * [terminal operators](https://kotlinlang.org/docs/reference/coroutines/flow.html#terminal-flow-operators) instead.
+     */
+    val members: Flow<ThreadUser>
+        get() = supplier.getThreadMembers(id)
+
+    /**
+     * Removes the user identified by [id] from the current thread.
+     * Requires the thread is not archived.
+     * Sending a message to the thread automatically unarchived it.
+     *
+     * @see [ThreadChannel.isArchived]
+     * @see [ThreadChannel.isLocked]
+     * @throws [RequestException] if something went wrong during the request.
+     */
+    suspend fun removeUser(userId: Snowflake) {
+        kord.rest.channel.removeUserFromThread(id, userId)
+    }
+
+    /**
+     * Adds the user identified by [id] from the current thread.
+     * Requires the thread is not archived.
+     *
+     * @see [ThreadChannel.isArchived]
+     * @see [ThreadChannel.isLocked]
+     * @throws [RequestException] if something went wrong during the request.
+     */
```

Comment:
Another requirement is that the bot needs to be able to send messages in a thread. Which should be `USE_PUBLIC_THREADS` || `USE_PRIVATE_THREADS` and `SEND_MESSAGES`, assuming this thread was retrieved via the API)

Reply:
That's not quite accurate, it turns out. From DMD:

![image](https://user-images.githubusercontent.com/204153/127344978-20f86db9-bc31-4780-987c-55dc44242b00.png)

Reply:
Can you elaborate on that? I assumed that disallowing `SEND_MESSAGES` or `USE_PUBLIC_THREADS` would prevent a user from sending messages in a thread. Do you need both disallowed to lock a user out?

Reply:
You need both disallowed, from what I've been able to glean from the threads channel on DMD. Either of them being enabled will allow users to interact in a thread, with unique effects more explicitly applying to the channel itself.

Reply:
Pink users on that server are Discord employees, by the way.

Reply:
That seems like a very counter-intuitive way to do things. I guess we'll hold off on documenting what exactly 'able to send messages' entails until we can test this more completely.

Reply:
That would probably be for the best